code-for-canada/project-thundercat
# [Add/Edit Email Responses/Tasks] Collapsing Item Container Titles Generator

# Description

I updated _Email.jsx_ component in order to populate the Collapsing Item Container titles, such as email response # and task #.

Even if we delete email(s), it recalculates the email response #. This functionality is only working for email responses, since the 'delete task' functionality is not implemented yet.

**Note:**

I was not able to simply use the _emailCount_ and _taskCount_ props to populate these titles, because it was giving me the wrong behavior:

![image](https://user-images.githubusercontent.com/23021242/55954949-1a3d8080-5c2e-11e9-8aaf-04a0b25ff7c4.png)

It repopulates all the Collapsing Item Containers on each state change, so it is taking the last _emailCount_ and _taskCount_ values to populate these titles.

## Type of change

- New feature (non-breaking change which adds functionality)

## Screenshot

**Example with some emails and tasks:**

![image](https://user-images.githubusercontent.com/23021242/55954758-8ff51c80-5c2d-11e9-81a6-b22e2fde7a35.png)

**After deleting the Email Response # 2:**

![image](https://user-images.githubusercontent.com/23021242/55973937-95b22880-5c54-11e9-9a4d-17bdb0be55ea.png)

# Testing

Manual steps to reproduce this functionality:

1.  Start the eMIB Sample Test.
2.  Complete the _pre-test_ process.
3.  Open the _Inbox_ tab.
4.  Add multiple email responses and tasks and make sure the the Collapsing Item Containers titles are right.

# Checklist

Applicable for all code changes.
